src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+        self.cross_modal_image_layernorm.apply(self._init_weights)
+
+        if config.link_tower_shared:
+            self.cross_modal_text_link_tower = LinkTower(config, self.tokenizer_config)
+            self.cross_modal_image_link_tower = LinkTower(config, self.tokenizer_config)
+        else:
+            self.cross_modal_text_link_tower = nn.ModuleList(
+                [LinkTower(config, self.tokenizer_config) for _ in range(config.num_hidden_layers - 1)]
+            )
+            self.cross_modal_image_link_tower = nn.ModuleList(
+                [LinkTower(config, self.tokenizer_config) for _ in range(config.num_hidden_layers - 1)]
+            )
+
+        self.cross_modal_text_link_tower.apply(self._init_weights)
+        self.cross_modal_image_link_tower.apply(self._init_weights)
+
+        # ===================== Freeze specified modules ===================== #
+
+        if config.freeze_ViT:
+            self.vit_model.requires_grad_(False)
+            if config.unfreeze_ViT_attention:
+                for name, param in self.vit_model.named_parameters():
+                    if "attn" in name:
+                        param.requires_grad_(True)
+            if config.unfreeze_ViT_layernorm:
+                for name, param in self.vit_model.named_parameters():
+                    if "ln_" in name:
+                        param.requires_grad_(True)
+
+        if config.freeze_RoBERTa:
+            self.text_transformer.requires_grad_(False)
+
+            if config.unfreeze_RoBERTa_embeddings:
+                self.text_transformer.embeddings.requires_grad_(True)
+
+            if config.unfreeze_RoBERTa_encoder:
+                self.text_transformer.encoder.requires_grad_(True)
+
+            if config.unfreeze_RoBERTa_attention:
+                for name, param in self.text_transformer.named_parameters():
+                    if "attention" in name:
+                        param.requires_grad_(True)
+
+            if config.unfreeze_RoBERTa_layernorm:
+                for name, param in self.text_transformer.named_parameters():
+                    if "LayerNorm" in name:
+                        param.requires_grad_(True)
+
+        if config.freeze_layer_count_roberta > 0:
+            modules = [
+                self.text_transformer.embeddings,
+                *self.text_transformer.encoder.layer[: config.freeze_layer_count_roberta],
+            ]  # Replace 5 by what you want
+            for module in modules:
+                for param in module.parameters():
+                    param.requires_grad = False
+
+        if config.freeze_layer_count_vit > 0:
+            modules = [*self.vit_model.visual.transformer.resblocks[: config.freeze_layer_count_vit]]
+            self.vit_model.visual.class_embedding.requires_grad = False
+            for module in modules:
+                # module.requires_grad_(False)
+                for param in module.parameters():
+                    param.requires_grad = False
```

Comment:
All these methods can be wrapper in a class method to make it cleaner ;)

Reply:
Handled! Thanks for the feedback.